src/Parsers.jl

```diff
@@ -170,6 +170,62 @@ Options(;
+    BoolConf{Bool}(options::Options=OPTIONS, refs=options.refs, trues=options.trues, falses=options.falses) = new{Bool}(refs, trues, falses)
+    BoolConf{Bool}(refs, trues, falses) = new{Bool}(refs, trues, falses)
+end
+struct DateTimeConf{T<:Union{DateTime,Date,Time}} <: AbstractDateTimeParseConf{T}
+    dateformat::Union{Nothing, Format}
+    # TODO: kwargs + validate inputs like Options does
+    DateTimeConf(::Type{T}, args...; kwargs...) where {T} = DateTimeConf{T}(args...; kwargs...)
+    DateTimeConf{T}(options::Options=OPTIONS, fmt=options.dateformat) where{T} = new{T}(fmt)
+    DateTimeConf{T}(fmt) where{T} = new{T}(fmt)
+end
+struct FloatConf{T<:Real} <: AbstractFloatParseConf{T}
+    decimal::UInt8
+    groupmark::Union{Nothing,UInt8}
+    # TODO: kwargs + validate inputs like Options does
+    FloatConf(::Type{T}, args...; kwargs...) where {T} = FloatConf{T}(args...; kwargs...)
+    FloatConf{T}(options::Options=OPTIONS, decimal=options.decimal, groupmark=options.groupmark) where {T} = new{T}(decimal, groupmark)
+    FloatConf{T}(decimal, groupmark) where {T} = new{T}(decimal, groupmark)
+end
+struct IntConf{T<:Integer} <: AbstractIntParseConf{T}
+    groupmark::Union{Nothing,UInt8}
+    # TODO: kwargs + validate inputs like Options does
+    IntConf(::Type{T}, args...; kwargs...) where {T} = IntConf{T}(args...; kwargs...)
+    IntConf{T}(options::Options=OPTIONS, groupmark=options.groupmark) where {T} = new{T}(groupmark)
+    IntConf{T}(groupmark) where {T} = new{T}(groupmark)
+end
+struct StringConf{T<:Union{AbstractString,Char,Symbol}} <: AbstractStringParseConf{T}
+    stripwhitespace::Bool
+    stripquoted::Bool
+    # TODO: kwargs + validate inputs like Options does
+    StringConf(::Type{T}, args...; kwargs...) where {T} = StringConf{T}(args...; kwargs...)
+    StringConf{T}(options::Options=OPTIONS, stripwhitespace=options.stripwhitespace, stripquoted=options.stripquoted) where {T} = new{T}(stripwhitespace, stripquoted)
+    StringConf{T}(stripwhitespace, stripquoted) where {T} = new{T}(stripwhitespace, stripquoted)
+end
+
+struct UnknownTypeConf{T} <: AbstractTypeParseConf{T} end
+
+conf(::Type{T},    opt_or_conf) where {T<:Integer}                           = IntConf{T}(opt_or_conf.groupmark)
+conf(::Type{T},    opt_or_conf) where {T<:Real}                              = FloatConf{T}(opt_or_conf.decimal, opt_or_conf.groupmark)
+conf(::Type{T},    opt_or_conf) where {T<:Union{DateTime,Date,Time}}         = DateTimeConf{T}(opt_or_conf.dateformat)
+conf(::Type{T},    opt_or_conf) where {T<:Union{AbstractString,Symbol,Char}} = StringConf{T}(opt_or_conf.stripwhitespace, opt_or_conf.stripquoted)
+conf(::Type{Bool}, opt_or_conf)                                              = BoolConf{Bool}(opt_or_conf.refs, opt_or_conf.trues, opt_or_conf.falses)
+conf(::Type{T},    opt_or_conf) where T                                      = UnknownTypeConf{T}()
```

Comment:
For custom types (like FixedPointDecimal), they need to define their own `CustomConf` struct, _and_ overload this `conf` function for to intercept their types, right?

Reply:
the `conf(...)` thing is only used internally so people can keep using e.g. `Int` instead of `IntConf()`. So it's not required for `FixedPointDecimal` is the user is okay with using `FixedDecimalConf()` as in input to `xparse`